test/distributed/test_inductor_collectives.py

```diff
@@ -191,6 +191,7 @@ def compile(func, example_inputs):
     @skip_if_lt_x_gpu(2)
     # TODO: somehow inductor bg compile threads are causing hangs at exit with distributed work dtor
     @patch.object(torch._inductor.config, "compile_threads", 1)
+    @patch.object(torch._dynamo.config, "dynamic_shapes", False)
```

Comment:
I am going to tell you what is going to happen. You are going to land this PR, and then get reverted because you broke someone.

I am not going to let you land this PR with any `dynamic_shapes` patches in test. Because you have disabled automatic dynamic and have assume static by default, you should have perfect parity with what we had before. If there is a discrepancy, it must be investigated first.